source/fundamentals/crud/write-operations/bulk.txt

```diff
@@ -288,14 +298,146 @@ operations to execute in any order:
+
+   ClientBulkWriteResult result = client.bulkWrite(bulkOperations);
+
+.. TODO: link documentation
+
+Replace Example
+~~~~~~~~~~~~~~~
+
+The following example shows how to use the ``bulkWrite()`` method to replace
+existing documents in the ``people`` and ``things`` collections.
+
+.. code-block:: java
+
+   MongoNamespace peopleNamespace = new MongoNamespace("db", "people");
+   MongoNamespace thingsNamespace = new MongoNamespace("db", "things");
+
+   List<ClientNamespacedWriteModel> bulkOperations = new ArrayList<>();
+
+   bulkOperations.add(ClientNamespacedWriteModel.replaceOne(
+         peopleNamespace, 
+         Filters.eq("_id", 1),
+         new Document("name", "Maximus Farquaad")
+      )
+   );
+
+   bulkOperations.add(ClientNamespacedWriteModel.replaceOne(
+         thingsNamespace, 
+         Filters.eq("_id", 1),
+         new Document("object", "potato")
+      )
+   );
+
+   ClientBulkWriteResult result = client.bulkWrite(bulkOperations);
+
+After this example runs successfully, the document in the ``people`` collection in 
+which the ``_id`` field has the value of ``1`` is replaced with a new
+document. The document in the ``things`` collection in which the 
+``_id`` field has the value of ``1`` is replaced with a new document.
+
+.. _java-sync-client-bulk-write-options:
+
+Bulk Write Options
+~~~~~~~~~~~~~~~~~~
+
+You can use the ``ClientBulkWriteOptions`` interface to specify options when
+running the ``bulkWrite()`` method.
+
+Order of Execution Example
+``````````````````````````
+
+By default, the write operations execute in the order that you specify them. However,
+you can pass ``false`` to the ``ordered()`` method on the ``ClientBulkWriteOptions``
+interface to perform write operations in an unordered way. When using the unordered
+option, an error-producing operation will not block execution of other bulk 
+write operations.
+
+The following code shows how to set the ``ordered()`` method when creating an
+instance of ``ClientBulkWriteOptions``.
+
+.. code-block:: java
+
+   MongoNamespace namespace = new MongoNamespace("db", "people");
+
+   ClientBulkWriteOptions options = new ClientBulkWriteOptions().ordered(false);
+
+   List<ClientNamespacedWriteModel> bulkOperations = new ArrayList<>();
+
+   bulkOperations.add(ClientNamespacedWriteModel.insertOne(namespace, new Document("_id", 1).append("name", "Donkey Kong)));
+   
+   // duplicate key
+   bulkOperations.add(ClientNamespacedWriteModel.insertOne(namespace, new Document("_id", 1).append("name", "Mario")));
+
+   bulkOperations.add(ClientNamespacedWriteModel.insertOne(namespace, new Document("name", "Princess Peach")));
+
+   ClientBulkWriteResult result = client.bulkWrite(bulkOperations, options);
+
+Even though the write operation inserting a document with a duplicate key will
+result in an error, the other operations will be executed because we have specified
+the ``ordered()`` option to be ``false``.
```

Comment:
```suggestion
Even though the write operation to insert a document that a duplicate key
results in an error, the other operations are committed because the operation is unordered.
```